core/libp2p/crypto/marshaller/key_marshaller_impl.cpp

```diff
@@ -60,33 +59,28 @@ namespace libp2p::crypto::marshaller {
     }
   }  // namespace
 
-  outcome::result<Buffer> KeyMarshallerImpl::marshal(
+  outcome::result<KeyMarshallerImpl::Buffer> KeyMarshallerImpl::marshal(
       const PublicKey &key) const {
     proto::PublicKey proto_key;
     OUTCOME_TRY(key_type, marshalKeyType(key.type));
     proto_key.set_key_type(key_type);
-
     proto_key.set_key_value(key.data.data(), key.data.size());
 
     auto string = proto_key.SerializeAsString();
-    Buffer out;
-    out.put(proto_key.SerializeAsString());
-
-    return out;
+    Buffer out(string.begin(), string.end());
+    return outcome::success(out);
   }
 
-  outcome::result<Buffer> KeyMarshallerImpl::marshal(
+  outcome::result<KeyMarshallerImpl::Buffer> KeyMarshallerImpl::marshal(
       const PrivateKey &key) const {
     proto::PublicKey proto_key;
     OUTCOME_TRY(key_type, marshalKeyType(key.type));
     proto_key.set_key_type(key_type);
     proto_key.set_key_value(key.data.data(), key.data.size());
 
     auto string = proto_key.SerializeAsString();
-    Buffer out;
-    out.put(proto_key.SerializeAsString());
-
-    return out;
+    Buffer out(string.begin(), string.end());
+    return outcome::success(out);
```

Comment:
Is necessary to explicitly return outcome::success()? Just `return Buffer(string.begin(), string.end());` won't work?